### PR TITLE
Addition of threading to MODIS download

### PR DESCRIPTION
**Pull request by DaveCasson**
Minor modifications to .py script and Jupyter Notebook.

### CODE REVIEW

File: 3b_parameters/MODIS_MCD12Q1_V6/1_download/download_modis_mcd12q1_v6.ipynb [hunk 1]
```diff
@@ -24,7 +24,9 @@
     "from netrc import netrc\n",
     "from pathlib import Path\n",
     "from shutil import copyfile\n",
-    "from datetime import datetime"
+    "from datetime import datetime\n",
+    "import concurrent\n",
+    "from concurrent.futures.thread import ThreadPoolExecutor"
```
Comment (wknoben) [inline 3b_parameters/MODIS_MCD12Q1_V6/1_download/download_modis_mcd12q1_v6.ipynb:29]: Is the line `import concurrent` needed?

File: 3b_parameters/MODIS_MCD12Q1_V6/1_download/download_modis_mcd12q1_v6.ipynb [hunk 2]
```diff
@@ -24,7 +24,9 @@
     "from netrc import netrc\n",
     "from pathlib import Path\n",
     "from shutil import copyfile\n",
-    "from datetime import datetime"
+    "from datetime import datetime\n",
+    "import concurrent\n",
```
Comment (wknoben) [inline 3b_parameters/MODIS_MCD12Q1_V6/1_download/download_modis_mcd12q1_v6.ipynb:28]: Can you update `requirements.txt` with the version of concurrent?

File: 3b_parameters/MODIS_MCD12Q1_V6/1_download/download_modis_mcd12q1_v6.ipynb [hunk 3]
```diff
@@ -233,74 +256,22 @@
...
+    "with ThreadPoolExecutor() as executor:\n",
+    "    futures = []\n",
+    "    for file_url_raw in file_list:\n",
+    "\n",
+    "        file_url = file_url_raw.strip()\n",
+    "\n",
+    "        file_name = file_url.split('/')[-1].strip()  # Get the last part of the url, strip whitespace and characters\n",
+    "        output_file = os.path.join(modis_path, file_name)\n",
+    "        print(f'Downloading file: {file_name} from: {file_url}')\n",
+    "        futures.append(executor.submit(request_get, file_url,output_file, usr, pwd))"
```
Comment (wknoben) [inline 3b_parameters/MODIS_MCD12Q1_V6/1_download/download_modis_mcd12q1_v6.ipynb:274]: For the files you did manage to download with the original code, does the new code give the same results?

**Review by wknoben — CHANGES_REQUESTED** [review]
Thanks for the suggestion. There are a few minor clarifications needed

File: 3b_parameters/MODIS_MCD12Q1_V6/1_download/download_modis_mcd12q1_v6.ipynb (hunk 2 above)
Comment (DaveCasson) [reply 3b_parameters/MODIS_MCD12Q1_V6/1_download/download_modis_mcd12q1_v6.ipynb:28]: This is path of the standard library, not needed for requirements.txt

File: 3b_parameters/MODIS_MCD12Q1_V6/1_download/download_modis_mcd12q1_v6.ipynb (hunk 3 above)
Comment (DaveCasson) [reply 3b_parameters/MODIS_MCD12Q1_V6/1_download/download_modis_mcd12q1_v6.ipynb:274]: It did appear so yes, though I did not do a file by file comparison. The files I downloaded did carry through successfully through the rest of the workflows.

File: 3b_parameters/MODIS_MCD12Q1_V6/1_download/download_modis_mcd12q1_v6.ipynb (hunk 1 above)
Comment (DaveCasson) [reply 3b_parameters/MODIS_MCD12Q1_V6/1_download/download_modis_mcd12q1_v6.ipynb:29]: Yes, it is. I had an issue before doing the import this way.

**Comment by DaveCasson** [comment]
Closing this pull request, as script has been updated and want to discuss.

**Pull request closed without merging**